File: 백준/Silver/11047. 동전 0/동전 0.py
import sys
readline = sys.stdin.readline
n, k = map(int, readline().split())
money = [int(readline()) for _ in range(n)]
money.reverse()
cnt = 0
for m in money:
    if k == 0:
        break
    if k / n == 0:
        continue
    else:
        cnt += k // m
        k = k % m
print(cnt)

File: 백준/Silver/11047. 동전 0/동전 0.swift
import Foundation

let nk = readLine()!.split { $0 == " " }.map { Int($0)! }
var money = [Int]()
for _ in 0..<nk[0] {
  money.append(Int(readLine()!)!)
}

money = money.reversed()
var k = nk[1]

var cnt = 0

for m in money {
  if k == 0 {
    break
  }

  if k / m == 0 {
    continue
  }
  cnt += k / m
  k = k % m
}

print(cnt)